File: packages/react-native-nitro-filetoolkit/ios/HybridFileToolkitFactory.swift
import NitroModules

final class HybridFileToolkitFactory: HybridFileToolkitFactorySpec {
  private let fileSystem = LazyModule<any HybridFileSystemSpec> {
    HybridFileSystem()
  }

  func getFileSystem() throws -> any HybridFileSystemSpec {
    fileSystem.get()
  }
}

File: packages/react-native-nitro-filetoolkit/android/src/main/java/com/margelo/nitro/filetoolkit/HybridFileToolkitFactory.kt
package com.margelo.nitro.filetoolkit

class HybridFileToolkitFactory : HybridFileToolkitFactorySpec() {
  private val fileSystem by lazy(LazyThreadSafetyMode.SYNCHRONIZED) { HybridFileSystem() }

  override fun getFileSystem(): HybridFileSystemSpec = fileSystem
}
